code/game/objects/items/robot/items/food.dm

```diff
@@ -209,20 +216,16 @@
 	damage = 10
 	embed_falloff_tile = 0
 
-/obj/projectile/bullet/reusable/lollipop/Initialize(mapload)
-	var/obj/item/food/lollipop/lollipop = new ammo_type(src)
-	color2 = lollipop.head_color
+/obj/projectile/bullet/lollipop/Initialize(mapload)
+	. = ..()
 	var/mutable_appearance/head = mutable_appearance('icons/obj/weapons/guns/projectiles.dmi', "lollipop_2")
-	head.color = color2
+	head.color = head_color = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
 	add_overlay(head)
-	return ..()
+	AddElement(/datum/element/projectile_drop, /obj/item/food/lollipop/cyborg)
+	RegisterSignal(src, COMSIG_PROJECTILE_ON_SPAWN_DROP, PROC_REF(handle_drop))
 
-/obj/projectile/bullet/reusable/lollipop/handle_drop()
-	if(!dropped)
-		var/turf/turf = get_turf(src)
-		var/obj/item/food/lollipop/lollipop = new ammo_type(turf)
-		lollipop.change_head_color(color2)
-		dropped = TRUE
+/obj/projectile/bullet/lollipop/proc/handle_drop(datum/source, obj/item/food/lollipop/lollipop)
```

Comment:
`SIGNAL_HANDLER`? you never know when someone might fuck with this and add more logic